x-pack/plugins/stack_alerts/server/rule_types/es_query/executor.ts

```diff
@@ -178,6 +180,15 @@ export async function executor(core: CoreSetup, options: ExecutorOptions<EsQuery
     });
 
     const id = alertId === UngroupedGroupId && !isGroupAgg ? ConditionMetAlertInstanceId : alertId;
+    const instances = alertId.split(',');
```

Comment:
This code seems very dependent on string parsing of potential user data.  What happens if one if the met terms includes `,`?  Feels like we should be using the values returned by the agg, instead of (I guess) parsing the result of us joining the values.  Maybe this is something new we need to start returning that we aren't today?

The test for `alertId !== UngroupedGroupId` also seems a little sketchy, since an alertId _could_ be `UngroupedGroupId`.  Seems like this code would work fine today, but is unneeded, and could potentially cause problems in the future.  Seems like you can just remove the  `alertId !== UngroupedGroupId` check completely.

Reply:
@pmuellr I changed the implementation in this commit: [da574d3](https://github.com/elastic/kibana/pull/191103/commits/da574d3a18e9538bcb9bac02301b9556b6988db2)

Is this good now?